### DISCUSSION TITLE
Hello, can you add a callback function that sets the hierarchical expansion and shrinkage icons of the tree widget?

### DISCUSSION THREAD

**Discussion by ddkwork** [cogentcore/core#825] [Q&A] [answered]
![Screenshot_20240119_004843_com.realvnc.viewer.android.jpg](https://github.com/cogentcore/core/assets/19886504/66e0b6aa-3267-4262-aa0f-8689ed57b030)

![Screenshot_20240119_005143_com.realvnc.viewer.android.jpg](https://github.com/cogentcore/core/assets/19886504/d915428f-af72-4146-8cb0-978c13b264b3)

https://www.naiveui.com/zh-CN/os-theme/components/tree

**Comment by ddkwork** [comment]
I've seen several UI libraries that provide this feature, can you support it? If you're busy dealing with other issues, I can try PR

https://ant.design/components/tree#components-tree-demo-switcher-icon

**Comment by kkoreilly** [comment]
Yes, it would be good to support this. I will get to this eventually, but you can do a PR if you want to speed things up. I would recommend adding the same three icon fields that are on switch to treeview (`IconOn`, `IconOff`, `IconUnk`), and then passing them when the switch icons are set (currently on line 363 of `treeview.go`). Also, for future reference, it would be better to file bugs and feature requests as issues instead of discussions if possible.

**Comment by ddkwork** [comment]
> Yes, it would be good to support this. I will get to this eventually, but you can do a PR if you want to speed things up. I would recommend adding the same three icon fields that are on switch to treeview (`IconOn`, `IconOff`, `IconUnk`), and then passing them when the switch icons are set (currently on line 363 of `treeview.go`). Also, for future reference, it would be better to file bugs and feature requests as issues instead of discussions if possible.

i will try, thx you.

**Comment by kkoreilly** [comment, marked as answer]
This has been implemented in #831.
